core-bundle/src/Resources/contao/templates/_new/content_element/code.html.twig

```diff
@@ -1,7 +1,7 @@
 {% extends "@Contao/content_element/_base.html.twig" %}
 {% use "@Contao/component/_stylesheet.html.twig" %}
 
-{% set highlighter_css_file = 'vendor/scrivo/highlight_php/styles/foundation.css' %}
+{% set highlighter_css_file = asset('vendor/scrivo/highlight_php/styles/foundation.css') %}
```

Comment:
It should use our `assets.packages` service, which already handles everything correctly:

https://github.com/contao/contao/blob/47697073752ae4d6fa6b63b28e0b54c50a8143f3/core-bundle/src/Resources/contao/library/Contao/Template.php#L452

@m-vo How do we add this to Twig?

Reply:
no, as I said, the **default** assets packages can be configured in your application. Like we do in https://github.com/contao-association/website/blob/main/config/config.yml#L27-L29 for example. This might influence the path to the public folder. See https://members.contao.org/layout/manifest.json for the files we use on that website, not sure what would happen if the path does not exist in the manifest.json

Reply:
Ok, so let‘s not make this more complicated than necessary. We do need a way to prefix the base path here and we sure don‘t want to use `env::base_path`. @contao/developers Suggestions?

Reply:
I think `app.request.base_path` should be available in twig?

```suggestion
{% set highlighter_css_file = app.request.base_path|default ~ '/vendor/scrivo/highlight_php/styles/foundation.css' %}
```

But I’m not sure if `app` always exists in all cases in our environment?

Reply:
It does and this works. 👍 Changed in ca951813c4d0bb25f63adba7aab8c54d9623674d.

Reply:
Using `app.request.base_path` is probably fine. :+1: I'd go with this. If we find a better solution later on, we can still change it.

Reply:
The correct solution would be `asset('foundation.css', 'scrivo/highlight.php')` after merging https://github.com/contao/contao/pull/4866
This will
 1. remove dependency on hardcoded path in templates
 2. automatically add a version strategy, so the link will be `foundation.css?v={version of composer package}`
 3. automatically prepend the root-page configured asset URL